src/mito2/src/memtable/partition_tree/shard_builder.rs

```diff
@@ -106,10 +109,8 @@ impl ShardBuilder {
             return Ok(None);
         }
 
-        let mut pk_to_index = BTreeMap::new();
-        let key_dict = self.dict_builder.finish(&mut pk_to_index);
-        let data_part = match &key_dict {
-            Some(dict) => {
+        let (data_part, key_dict) = match self.dict_builder.finish() {
+            Some((dict, pk_to_index)) => {
```

Comment:
So here is the fix, use `pk_to_index` from `dict_builder` instead of construct one here, right?

Reply:
> So here is the fix, use `pk_to_index` from `dict_builder` instead of construct one here, right?

No, here's just to avoid creating a new BTreeMap instance.